src/contract_wrappers/modules/sto/__tests__/capped_sto_wrapper.test.ts

```diff
@@ -272,6 +273,188 @@ describe('CappedSTOWrapper', () => {
+      // Address expected
+      const expectedAllowBeneficialInvestmentResult = true;
+      // Mocked method
+      const mockedAllowBeneficialInvestmentMethod = mock(MockedCallMethod);
+      // Stub the method
+      when(mockedContract.allowBeneficialInvestments).thenReturn(instance(mockedAllowBeneficialInvestmentMethod));
+      // Stub the request
+      when(mockedAllowBeneficialInvestmentMethod.callAsync()).thenResolve(expectedAllowBeneficialInvestmentResult);
+
+      // Owner Address expected
+      const expectedOwnerResult = '0x5555555555555555555555555555555555555555';
+
+      // Mock web3 wrapper owner
+      when(mockedWrapper.getAvailableAddressesAsync()).thenResolve([expectedOwnerResult]);
+
+      // Pause Address expected
+      const expectedPausedResult = false;
+      // Mocked method
+      const mockedPausedMethod = mock(MockedCallMethod);
+      // Stub the method
+      when(mockedContract.paused).thenReturn(instance(mockedPausedMethod));
+      // Stub the request
+      when(mockedPausedMethod.callAsync()).thenResolve(expectedPausedResult);
+
+      // Address expected
+      const expectedFundRaiseTypeResult = true;
+      // Mocked method
+      const mockedFundRaiseMethod = mock(MockedCallMethod);
+      // Stub the method
+      when(mockedContract.fundRaiseTypes).thenReturn(instance(mockedFundRaiseMethod));
+      // Stub the request
+      when(mockedFundRaiseMethod.callAsync(FundRaiseType.ETH)).thenResolve(expectedFundRaiseTypeResult);
+
+      const expectedStartTimeResult = new BigNumber(1735689000);
+      // Mocked method
+      const mockedStartTimeMethod = mock(MockedCallMethod);
+      // Stub the method
+      when(mockedContract.startTime).thenReturn(instance(mockedStartTimeMethod));
+      // Stub the request
+      when(mockedStartTimeMethod.callAsync()).thenResolve(expectedStartTimeResult);
+
+      const expectedEndTimeResult = new BigNumber(1735689600);
+      // Mocked method
+      const mockedEndTimeMethod = mock(MockedCallMethod);
+      // Stub the method
+      when(mockedContract.endTime).thenReturn(instance(mockedEndTimeMethod));
+      // Stub the request
+      when(mockedEndTimeMethod.callAsync()).thenResolve(expectedEndTimeResult);
+
+      // ****** Real method setup and call *****
+      const txPayableData = {
+        ...{},
+        from: expectedOwnerResult,
+        value: new BigNumber(1),
+      };
+      const mockedParams = {
+        beneficiary: expectedOwnerResult,
+        value: new BigNumber(1),
+        txData: txPayableData,
+        safetyFactor: 10,
+      };
+      const expectedResult = getMockedPolyResponse();
+      // Mocked method
+      const mockedMethod = mock(MockedSendMethod);
+      // Stub the method
+      when(mockedContract.buyTokens).thenReturn(instance(mockedMethod));
+      // Stub the request
+      when(
+        mockedMethod.sendTransactionAsync(mockedParams.beneficiary, txPayableData, mockedParams.safetyFactor),
```

Comment:
Problem is the same we discussed recently. `txPayable` data is a different instance here and in the real call. You should use `objectContaining` instead.

Reply:
Okay used objectContaining(txPayable) seems to work